src/AzureOpenAIProxy.Management/Services/EventService.cs

```diff
@@ -8,9 +8,19 @@
 
 namespace AzureOpenAIProxy.Management.Services;
 
-public class EventService(IAuthService authService, AoaiProxyContext db) : IEventService, IDisposable
+
+public class EventService() : IEventService, IDisposable
 {
-    private readonly DbConnection conn = db.Database.GetDbConnection();
+    private readonly AoaiProxyContext db = null!;
+    private readonly DbConnection conn = null!;
+    private readonly IAuthService authService = null!;
+
+    public EventService(IAuthService authService, IDbContextFactory<AoaiProxyContext> dbContextFactory) : this()
+    {
+        this.authService = authService ?? throw new ArgumentNullException(nameof(authService));
+        db = dbContextFactory.CreateDbContext() ?? throw new ArgumentNullException(nameof(dbContextFactory));
+        conn = db.Database.GetDbConnection() ?? throw new ArgumentNullException(nameof(dbContextFactory));
+    }
```

Comment:
I don't understand the point of this refactor - none of the parameters are nullable